zk/stages/stage_interhashes.go

```diff
@@ -165,21 +156,12 @@ func SpawnZkIntermediateHashesStage(s *stagedsync.StageState, u stagedsync.Unwin
 		expectedRootHash := syncHeadHeader.Root
 		headerHash := syncHeadHeader.Hash()
 		if root != expectedRootHash {
-			if useBatch {
-				eridb.RollbackBatch()
-			}
```

Comment:
Due to the changes in the increment function the batch will have already been committed before we've checked the root.  I think we should still handle rolling back and committing the batch to the parent call here.